internal/server/data/identity.go

```diff
@@ -22,15 +24,16 @@ func (i identitiesTable) Table() string {
 }
 
 func (i identitiesTable) Columns() []string {
-	return []string{"created_at", "created_by", "deleted_at", "id", "last_seen_at", "name", "organization_id", "updated_at", "verification_token", "verified"}
+	return []string{"created_at", "created_by", "deleted_at", "id", "last_seen_at", "name", "organization_id", "ssh_username", "updated_at", "verification_token", "verified"}
 }
 
 func (i identitiesTable) Values() []any {
-	return []any{i.CreatedAt, i.CreatedBy, i.DeletedAt, i.ID, i.LastSeenAt, i.Name, i.OrganizationID, i.UpdatedAt, i.VerificationToken, i.Verified}
+	sshUsername := sql.NullString{String: i.SSHUsername, Valid: i.SSHUsername != ""}
+	return []any{i.CreatedAt, i.CreatedBy, i.DeletedAt, i.ID, i.LastSeenAt, i.Name, i.OrganizationID, sshUsername, i.UpdatedAt, i.VerificationToken, i.Verified}
 }
 
 func (i *identitiesTable) ScanFields() []any {
-	return []any{&i.CreatedAt, &i.CreatedBy, &i.DeletedAt, &i.ID, &i.LastSeenAt, &i.Name, &i.OrganizationID, &i.UpdatedAt, &i.VerificationToken, &i.Verified}
+	return []any{&i.CreatedAt, &i.CreatedBy, &i.DeletedAt, &i.ID, &i.LastSeenAt, &i.Name, &i.OrganizationID, (*optionalString)(&i.SSHUsername), &i.UpdatedAt, &i.VerificationToken, &i.Verified}
```

Comment:
There's a new pattern here for handling of possibly null values. 

In `Values` we use `sql.NullString` which allows us to save a null to the database if `SSHUsername` is the empty string.

In` ScanFields` we use a wrapper around `sql.NullString` to allow reading a null value as the empty string.

This lefts us avoid ` *string` field while still supporting nulls in the database.

Nulls in the database here are important because there's a unique constraint on this field. Multiple null values are allowed, but only a single row would allow the empty string. 

This is being done so that we don't have to generate every username immediately on deploy. We can generate a username as users login or add public keys.

We could avoid this by generating all the usernames in a migration on deploy, but I'm not sure if that's a good option.

Reply:
We could avoid this new type by assigning the SSH username in the migration, correct?

Reply:
Ya, I think we can avoid this if we migrate all users immediately